src/structures/Message.js

```diff
@@ -241,7 +241,9 @@ class Message extends Base {
    */
   patch(data) {
     const clone = this._clone();
-    this._edits.unshift(clone);
+    const { messageEditHistoryMaxSize } = this.client.options;
+    const editsLimit = messageEditHistoryMaxSize === -1 ? Infinity : messageEditHistoryMaxSize;
+    if (this._edits.unshift(clone) > editsLimit) this._edits.pop();
```

Comment:
Just asking, I may be wrong about the idea, but why not skip the caching if the provided max size is `0` that way you aren't pushing the item into the array then instantly popping it back out?